packages/style-engine/class-wp-style-engine-css-rules-store.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * WP_Style_Engine_CSS_Rules_Store
+ *
+ * A store for WP_Style_Engine_CSS_Rule objects.
+ *
+ * @package Gutenberg
+ */
+
+if ( class_exists( 'WP_Style_Engine_CSS_Rules_Store' ) ) {
+	return;
+}
+
+/**
+ * Holds, sanitizes, processes and prints CSS declarations for the style engine.
+ *
+ * @access private
+ */
+class WP_Style_Engine_CSS_Rules_Store {
+
+	/**
+	 * An array of named WP_Style_Engine_CSS_Rules_Store objects.
+	 *
+	 * @static
+	 *
+	 * @var WP_Style_Engine_CSS_Rules_Store[]
+	 */
+	protected static $stores = array();
+
+	/**
+	 * An array of CSS Rules objects assigned to the store.
+	 *
+	 * @var WP_Style_Engine_CSS_Rule[]
+	 */
+	protected $rules = array();
+
+	/**
+	 * Get an instance of the store.
+	 *
+	 * @param string $store_name The name of the store.
+	 *
+	 * @return WP_Style_Engine_CSS_Rules_Store
+	 */
+	public static function get_store( $store_name ) {
+		if ( ! isset( static::$stores[ $store_name ] ) ) {
+			static::$stores[ $store_name ] = new WP_Style_Engine_CSS_Rules_Store( $store_name );
```

Comment:
Do we need to pass `$store_name` to the constructor?

Also I think changing this to `new static()` would mean that any class extending this one will be referring to the correct object. I think... 😄

Reply:
> Do we need to pass `$store_name` to the constructor?

We want to be able to use multiple stores, so we can have one for global styles, one for theme-styles etc. By using named stores we can easily differentiate between them and call them at any point in the execution path to add things or otherwise manipulate the stores.

I went ahead and added a `default` value though, so we can now call the constructor without a `$store_name` to reference the default store 👍 

> Also I think changing this to `new static()` would mean that any class extending this one will be referring to the correct object. I think... 😄

Good call! Changed it to `static()` 👍